File: MyClub/MyClub.Model/SearchObjects/BaseSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace MyClub.Model.SearchObjects
{
    public class BaseSearchObject
    {
        public string? FTS { get; set; }
        public int? Page { get; set; } = 0;
        public int? PageSize { get; set; } = 10;
        public bool IncludeTotalCount { get; set; } = true;
        public bool RetrieveAll { get; set; } = false;
    }
} 

File: MyClub/UI/myclub_mobile/lib/models/search_objects/base_search_object.dart
class BaseSearchObject {
  String? fts;
  int? page;
  int? pageSize;
  bool includeTotalCount;
  bool retrieveAll;

  BaseSearchObject({
    this.fts,
    this.page = 0,
    this.pageSize = 10,
    this.includeTotalCount = true,
    this.retrieveAll = false,
  });

  Map<String, dynamic> toJson() {
    return {
      'fts': fts,
      'page': page,
      'pageSize': pageSize,
      'includeTotalCount': includeTotalCount,
      'retrieveAll': retrieveAll,
    };
  }
}
